main.js

```diff
@@ -19,4 +19,26 @@ var outData = {};
 inData.forEach((message) => {
   outData[message["id"]] = message["defaultMessage"];
 });
-fs.writeFileSync(process.argv[3], JSON.stringify(outData, null, 2));
+
+if (process.argv[3] === "--comments") {
+  process.stdout.write("generating bash scripts...\n");
+  var messageInfo = JSON.parse(fs.readFileSync(__dirname + "/bash_scripts/hashmap.json"));
+  var scriptPath = __dirname + "/bash_scripts/put_comments.sh";
+  fs.writeFileSync(scriptPath, "#!/bin/bash\n");
+  fs.writeFileSync(scriptPath, "set -x\n");
+  inData.forEach((message) => {
+    var info = messageInfo.find(mi => mi.key == message.id);
+    if (info) {
+      var header = " -H \"Content-Type: application/json\"";
+      var escapedDescription = message.description.replace(/(["])/g, "\\\\\\$1");
```

Comment:
I don't want to talk about it.

If you must know, this takes care of escaping `"` characters that may exist in comments to be uploaded. The first arg to `replace` is just a regex saying `grab all instances of " and put each in a capture group`. The second replaces each with the esoteric `\\\\\\$1`, which I should explain  further.

For starters, remember that this is javascript writing bash. So after the first layer of replacement (js -> bash, javascript processes `\\` into an unescaped `\` and replaces `$1` with our capture group), the bash file will contain `\\\"` instead of `"`. One more layer of replacement (bash -> curl) takes place when bash executes the curl statement, so curl sees `\"`. Since we use `"` to delimit curl's `--data` argument, it knows to handle that escaped `"` correctly when sending across the wire.

... phew